TheMovieDatabase/Services/ProfileService.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  ProfileService.swift
+//  TheMovieDatabase
+//
+//  Created by Evgeny Syrtsov on 4/17/20.
+//  Copyright © 2020 Evgeny Syrtsov. All rights reserved.
+//
+
+import Foundation
+
+final class ProfileService {
+
+    // MARK: - Properties
+
+    private let moviesLoadingService = MoviesLoadingService()
+    private let guestAccount = Account(id: nil,
+                                       name: "Guest",
+                                       username: "Guest",
+                                       includeAdult: true)
+
+    // MARK: - Methods
+
+    func getAccountDetails(completion: @escaping (Account?) -> Void) {
+        if AuthorizationService.getSessionId() != nil {
+            guard
+                let sessionId = AuthorizationService.getSessionId(),
+                let url = URL(string: UrlParts.baseUrl + "account")?
+                    .appending("api_key", value: UrlParts.apiKey)?
+                    .appending("session_id", value: sessionId)
+            else {
+                return
+            }
+            URLSession.shared.dataTask(with: url) { (data, response, error) in
+                let decoder = JSONDecoder()
+                decoder.keyDecodingStrategy = .convertFromSnakeCase
+                guard let data = data else {
+                    return
+                }
+                do {
+                    let result = try decoder.decode(Account.self, from: data)
+                    DispatchQueue.main.async {
+                        completion(result)
+                    }
+                } catch {
+                    completion(nil)
+                }
+            }.resume()
+        } else {
+            completion(guestAccount)
+        }
+    }
+
+    func setFavoriteTo(_ isFavorite: Bool,
```

Comment:
I don't think that profile service should set favorites for movies, do you?

Reply:
Url of this method starting with baseUrl + /account so it’s work with account’s data, that’s why this method to set like/dislike is here